src/components/ToggleButton.js

```diff
@@ -0,0 +1,36 @@
+import React, {PropTypes} from 'react'
+import styles from './ToggleButton.css'
+import CSSModules from 'react-css-modules'
+import {changeTokenType} from '../actions/actions.js'
+import getID from '../utils/getID'
+
+class ToggleButton extends React.Component {
+  componentWillMount () {
+    this.state = {isChecked: false}
+    this.onChange = this.onChange.bind(this)
+    this.toggleId = getID()
+  }
+  onChange (e) {
+    let nextVal = !this.state.isChecked
+    this.setState({
+      isChecked: nextVal
+    })
+    // we need to inverse the nextVal because checked means isPreview === false
```

Comment:
the previous version was not wrong since I am setting the `state.isChacked`and then negate it .